{{ cookiecutter.formal_name }}/app/src/main/java/org/beeware/android/MainActivity.java

```diff
@@ -136,6 +161,12 @@ private void unpackPython(Map<String, File> paths) throws IOException {
         File userCodeDir = paths.get("user_code");
         Log.d(TAG, "Unpacking Python assets to base dir " + userCodeDir.getAbsolutePath());
         unpackAssetPrefix(getAssets(), "python", userCodeDir);
+        if (actualLastUpdateTime != null) {
```

Comment:
Is this correct? If we've unpacked, it doesn't matter what the actualLastUpdateTime is - we should be updating the storedLastUpdateTime.

Reply:
IMHO if the last update time is null, then I don't want to write it, for two reasons.

1. I have to create a `String` out of `null`, just to learn that it's four bytes long. Ugh, that'd be dumb boilerplate.

2. If it's null, then that means we can't actually compute last update time value, which means that this cache-validation strategy has failed. We can't verify next startup that the app is unchanged. Therefore, I prefer not to write anything to the file.

Reply:
👍 That makes sense.